base/Makefile

```diff
@@ -149,7 +155,40 @@ clean:
 
 .deploy: .deployment .cluster-config .network
 
-.network: .gen/k8s | .cluster-config
+define CERTIFICATE_TEMPLATE
+apiVersion: "networking.gke.io/v1beta1"
+kind: "ManagedCertificate"
+metadata:
+  name: "kctf-certificate"
+spec:
+  domains:
+endef
+
+define INGRESS_TEMPLATE
+apiVersion: "networking.k8s.io/v1beta1"
+kind: "Ingress"
+metadata:
+  name: "https"
+  annotations:
+    networking.gke.io/managed-certificates: "kctf-certificate"
+spec:
+  backend:
+    serviceName: "chal"
+    servicePort: 1
+endef
+
+export CERTIFICATE_TEMPLATE
+export INGRESS_TEMPLATE
+
+.https: .cluster-config
+ifeq ($(HTTPS), true)
+	(echo "$${CERTIFICATE_TEMPLATE}"; echo '    -' "$(CHALLENGE_NAME).$(DOMAIN_NAME)") | kubectl apply -f -
```

Comment:
This command would fail if DOMAIN_NAME is not set. Should we check and inform user to save her/him troubleshooting time? (I need to do the same with DNS btw :)

Reply:
good point, I'll update this CL to be based on yours once you submit yours